### PR TITLE
refactor: replace supabase/ui in design system typography components

### LINKED ISSUE #1133: Refactor: Replace supabase typography with native paragraph
### Type of feature

🧑‍💻 Refactor

### Current behavior

Based on #338 this issue should replace Supabase UI typography with Radix UI but Radix UI doesn't have a text component. So I think we can replace it with a native paragraph.
Thoughts? 
cc @0-vortex @OgDev-01 @Deadreyo

### Suggested solution

The component can be found here -> https://github.com/open-sauced/insights/blob/beta/components/atoms/Typography/text.tsx

### Additional context

We can also replace the [Title](https://github.com/open-sauced/insights/blob/beta/components/atoms/Typography/title.tsx) component with a native HTML header as well in the same fix.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by babblebey**
## Description

This pull request aims to refactor the typography components (`Title` and `Text`) in the design system to improve consistency, maintainability, and extensibility. The changes in this branch replaces `supabase/ui` typography component with native `html` alternatives leveraging `@tailwindcss/typography` and introduce enhancements to provide flexibility.

### Introduce the `TypographyWrapper` component

This component serves as a wrapper around child components exposing `children` components to the `prose` utility class from `@tailwindcss/typography` which adds beautiful typographic defaults to uncontrollable vanilla HTML, like HTML rendered from Markdown. 

It also serves as a primary wrapper for the `Title` and `Text` components.

#### Implementation

```js
const TypographyWrapper: React.FC<TypographyWrapperProps> = ({ children, className, size = "sm", ...props }) => {
  return (
    <span className={clsx(
      "prose", 
      `prose-${size}`, 
      className)} 
      {...props}
    >
      {children}
    </span>
  );
};
```

- The Component accepts the `className` and `size` props which uses as/to modify the below `class`es for the returned element 
  - `"prose"`: This is a default CSS class that provides basic typographic styles, such as margins and line heights, to improve readability of the text.
  - `"prose-${size}"`: This class is dynamically generated based on the `size` prop. For example, if size is set to "sm", it will apply the class "prose-sm", which provides smaller font size and line height. Similarly, for other size values, appropriate styles will be applied.
  - `className`: This is used to apply any additional CSS classes passed through the `className` prop.

- The component also passes any other props that might be provided to the `TypographyWrapper` through the `...props` spread operator. This allows developers to use additional attributes on the wrapper element.

#### Use Cases

- As Primary Wrapper for `Title` components.

```js
const Title: React.FC<TitleProps> = ({ children, level = 1, weight = "medium", className, ...props }) => {
  const TitleTag = `h${level}` as keyof JSX.IntrinsicElements;

  return (
    <TypographyWrapper> // <- 🍕🍕🍕🍕🍕
      <TitleTag className={clsx(className, `font-${weight}`)} {...props}>
        {children}
      </TitleTag>
    </TypographyWrapper> // <- 🍕🍕🍕🍕🍕
  );
};
```

- Useable sparingly (maybe for `Blogposts` highlight type 🍕)
```js
<TypographyWrapper>
  <p>We get lots of complaints about it actually, with people regularly asking us things like:</p>
  <blockquote>
    <p>Why is Tailwind removing the default styles on my <code>h1</code> elements? How do I disable this? What do you mean I lose all the other base styles too?</p>
  </blockquote>
</TypographyWrapper>
```
screenshot-localhost_3000-2023 08 01-15_42_44

## Other Changes

- `Text` component very much follows similar implementation and integrates exactly as that in the initial set-up without regression.

```js
<Text>
  Normal Text
</Text>

<Text className="!text-2xl !text-red-500">
  Normal Text (with ClassNames)
</Text>

<Text keyboard>
  Keyboard Text
</Text>

<Text code>
  Code Text
</Text>

<Text underline>
  Underlined Text
</Text>

<Text strikethrough>
  Strikethrough Text
</Text>

<Text small>
  Small Text
</Text>
```

screenshot-localhost_3000-2023 08 01-16_14_02

- `Title` component also follows similar implementation and integrates exactly as that in the initial set-up.

```js
<Title>
  Title Default Level and Weight
</Title>

<Title level={1} weight="light">
  Title Level 1 with light weight
</Title>

<Title level={3} className="!text-yellow-400">
  Title Level 3 with className
</Title>
```

screenshot-localhost_3000-2023 08 01-16_33_55

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [x] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #1133 

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

Install `@tailwindcss/typography` for local development

### CODE REVIEW

**Comment by OgDev-01** [comment]
Hey @babblebey, is this ready for review?

**Comment by babblebey** [comment]
> is this ready for review?

Almost 🤞

**Review by OgDev-01 — APPROVED** [review]
LGTM... Thanks for taking a step forward and adding the tailwinds typography plugin 🍕 

Quite a well crafted novel on the `typography-wrapper` story though 😁

File: components/atoms/Typography/text.tsx [hunk 1]
```diff
@@ -1,18 +1,70 @@
+/* eslint-disable prettier/prettier */
```
Comment (brandonroberts) [inline components/atoms/Typography/text.tsx:1]: Why are we disabling prettier here?

Comment (brandonroberts) [reply components/atoms/Typography/text.tsx:1]: Ok, if the linter and formatter are clashing, we should fix that in a follow-up

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**